storage/remote/queue_manager.go

```diff
@@ -629,14 +629,8 @@ func processExternalLabels(ls labels.Labels, externalLabels labels.Labels) label
 			j++
 		}
 	}
-	for ; i < len(ls); i++ {
-		result = append(result, labels.Label{
-			Name:  ls[i].Name,
-			Value: ls[i].Value,
-		})
-	}
-	result = append(result, externalLabels[j:]...)
-	return result
+	result = append(result, ls[i:]...)
+	return append(result, externalLabels[j:]...)
```

Comment:
What about simplfying further? 

```suggestion
	return append(append(result, ls[i:]...), externalLabels[j:]...)
```

Reply:
nice, so simple.